.github/workflows/build-packages.yml

```diff
@@ -168,27 +193,33 @@ jobs:
       - name: Upload artifact
         uses: actions/upload-artifact@v2
         with:
-          name: bob-zip
+          name: bob-zip-${{ matrix.bob_key_size }}
           path: bob_${{ env.VERSION }}.zip
 
   upload-zip:
     runs-on: ubuntu-latest
     needs: zip-file
     environment: prod
+    strategy:
+      matrix:
+        bob_key_size: [8, 16]
+        include:
+          - bob_key_size: 16
+            asset_name_postfix: _key16
```

Comment:
Same here: You can use conditional step that generates this name and save it to environment variable only when bob_key_size != 8